cpp/arcticdb/util/test/test_storage_lock.cpp

```diff
@@ -182,6 +182,46 @@ struct ForceReleaseLockTask {
     }
 };
 
+struct OptimisticForceReleaseLockTask {
+    std::shared_ptr<LockData> data_;
+    size_t timeout_ms_;
+    size_t retry_ms_;
+
+    OptimisticForceReleaseLockTask(std::shared_ptr<LockData> data, size_t timeout_ms, size_t retry_ms) :
+            data_(std::move(data)),
+            timeout_ms_(timeout_ms),
+            retry_ms_(retry_ms)
+    {
+    }
+
+    folly::Future<folly::Unit> operator()() {
+        std::this_thread::sleep_for(std::chrono::milliseconds(1));
+        StorageLock<> lock{data_->lock_name_};
+        bool contended = true;
+        size_t total_wait = 0;
+        while (contended && total_wait < timeout_ms_) {
+            contended = !lock.try_lock(data_->store_);
+            std::this_thread::sleep_for(std::chrono::milliseconds(retry_ms_));
+            total_wait += retry_ms_;
+
+        }
+        if (contended) {
+            data_->timedout_ = true;
+            data_->contended_ = true;
+        } else {
+            ++data_->vol_;
+            // This should be unnecessary as we are already locked
+            std::lock_guard l{data_->mutex_};
```

Comment:
`LockData::atomic_` is of type `std::atomic<uint64_t>` (which is provides the strongest memory ordering (Sequentially-consistent ordering)). There is not need to guard touching this variable with a lock. It is guaranteed by the standard that no data race is happening.

Reply:
Please see my comment below about scrapping this mutex.